bleep-cli/src/scala/bleep/sbtimport/buildFromBloopFiles.scala

```diff
@@ -161,7 +161,8 @@ object buildFromBloopFiles {
         `source-layout` = Some(sources.sourceLayout),
         `sbt-scope` = Some(projectType.sbtScope),
         isTestProject = if (projectType.testLike) Some(true) else None,
-        testFrameworks = testFrameworks
+        testFrameworks = testFrameworks,
+        `source-generator-scripts` = model.JsonList.empty // TODO can we do better ?
```

Comment:
Absolutely! 

Let me explain what is in place today:

after importing the full build from sbt, bleep looks at whatever generated (re-)sources are present.

this is done in [generateBuild](https://github.com/oyvindberg/bleep/blob/master/bleep-cli/src/scala/bleep/sbtimport/generateBuild.scala#L49).

If there are any, it generates a `GenerateResources` script, which can re-generate all those files. 

Obviously this approach is not-clever, since it just writes strings back to disk.

It is the intention that the user replace the implementation of the script with calls to code which can re-generate the same things. if you have generated protobuf you'll call protobuf yourself. My hope is that we can port a rather small set of sbt plugins to enable most usage of generated resources in sbt projects in bleep. Potentially we could extract for instance protobuf configuration from the sbt build, if we wanted to invest time in that.


A path forward here is that we take what is in place today, group these files by project, and generate one resource-generating script per project. Possibly with some extra logic for cross projects to make the number of generated scripts lower by using the projectName parameter discussed in the other thread.

If the rest of the PR ends up being a lot of work we can absolutely do this in a followup PR.

Reply:
> If the rest of the PR ends up being a lot of work we can absolutely do this in a followup PR. 

Right, I think I'll leave that aside as I'm not a fan of the current approach : it gives an initial impression that bleep does something really smart / magic.

Reply:
What i described above is a migration aid when coming from sbt to get a working build immediately.

That seems magical to you?

Reply:
It is, in the sense that without looking at the sources of the generated scripts, I'll get a false sense that the script will copy the behaviour of SBT's or mill's code-generators. 

When I say "I", I'm projecting myself as a random user. I personally have a pretty good understanding of build tools for having contributed to mill (in a non trivial manner), so by curiosity I did open the generated scripts to understand how it was working. A random user might just think "oh, cool, compilation works so my protobuf codegen logic has been ported over, neat", and be in for a puzzlement or a disappointment when they discover it's not actually copying the logic but rather the outputs.

Reply:
Yeah, makes sense.

In an attempt to communicate this the build import will print [this warning](https://github.com/oyvindberg/bleep/blob/master/bleep-cli/src/scala/bleep/sbtimport/generateBuild.scala#L87) in red after importing, and the script itself will print [this warning](https://github.com/oyvindberg/bleep/blob/master/snapshot-tests/bloop/imported/scripts/src/scala/scripts/GenerateResources.scala#L10) when it runs.

Reply:
And like i said elsewhere, we could extend the import to recognize common source generation plugins like `sbt-boilerplate` and actually set it up properly :)

Reply:
Yup that's fair, hadn't noticed the warning (or had forgotten about it), I've not tried to import an sbt project in a while. 

TBH my opinion on this should be taken with a grain of salt (I 100% respect the fact that OSS authors are entitled to driving the design however they want), and I'm not even saying the feature shouldn't exist, but rather that the feature doesn't interest me and therefore my motivation to work on that part is nil 😅

Reply:
Haha fine, don't worry about it :)

For me it's crucial that there is a a one-command import which produces a working build (goal 3) so new users can test bleep in real conditions. Only after trying it out do you have to invest time into resource generation scripts.

Reply:
I'm resolving the conversations, just leaving this open as a todo my myself after merge